wordpress/wp-content/plugins/memberful-wp/src/user/role_decision.php

```
@@ -42,6 +46,40 @@ public function role_for_user($current_role, $current_subscriptions) {
+  /**
+   * Determine the role for a user based on their plan subscriptions
+   * @param array $current_subscriptions User's current subscription plans
+   * @return string The role to assign
+   */
+  private function role_for_user_with_plan_mappings( $current_subscriptions ) {
+    $plan_mappings = memberful_wp_get_all_plan_role_mappings();
+    
+    // Find the highest priority role based on user's subscriptions
+    $assigned_roles = array();
+    
+    foreach ( $current_subscriptions as $plan_id => $subscription_data ) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$subscription_data'. (undefined)

(UnusedLocalVariable)